californium-core/src/main/java/org/eclipse/californium/core/network/InMemoryMessageExchangeStore.java

```diff
@@ -353,12 +325,8 @@ public boolean registerOutboundResponse(final Exchange exchange) {
 	 * Purges all registered exchanges from this store.
 	 */
 	public void clear() {
-		synchronized (messageIdProvider) {
-			synchronized (exchangesByToken) {
-				exchangesByMID.clear();
-				exchangesByToken.clear();
-			}
-		}
+		exchangesByMID.clear();
```

Comment:
IMHO we need to keep synchronization here in order to prevent cases where thread A invokes `registerOutbound` and thread B invokes `clear` and any of the method invocations in `clear` are interwoven with the registration invocations in `registerOutbound`, don't you think?